src/struct_ls/HYPRE_struct_int.c

```diff
@@ -35,55 +36,36 @@ hypre_StructVectorSetRandomValues( hypre_StructVector *vector,
    hypre_SetIndex3(unit_stride, 1, 1, 1);
 
    boxes = hypre_StructGridBoxes(hypre_StructVectorGrid(vector));
+
+   if (hypre_GetExecPolicy1(memory_location) == HYPRE_EXEC_DEVICE)
```

Comment:
Is there a reason to not use cuRAND/rocRAND? Just curious, this question is out of scope for this PR

Reply:
It is just to have exactly the same result on GPUs as CPUs. We thought it would be important to have identical results when developed GPU structured solvers. Maybe we can add an option to use curand later. Good point!